src/components/triniti-app-nav/MainNavContent.jsx

```diff
@@ -31,6 +54,21 @@ class MainNavContent extends React.Component {
     }
   }
 
+  handleShowModal(modal) {
+    const ModalComponent = createLazyComponent(modal);
+    this.setState({
+      showModal: true,
+      modal: <ModalComponent onHide={this.handleHideModal} />,
```

Comment:
however we end up using onShow/onHide, the modals definitely need a toggle prop because that is what allows pressing esc to close it, among other things